src/Renderers/TitleNodeRenderer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Docs Builder package.
+ * (c) Ryan Weaver <ryan@symfonycasts.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SymfonyDocsBuilder\Renderers;
+
+use Doctrine\RST\Environment;
+use Doctrine\RST\Nodes\TitleNode;
+use Doctrine\RST\Renderers\NodeRenderer;
+use Doctrine\RST\Templates\TemplateRenderer;
+
+class TitleNodeRenderer implements NodeRenderer
+{
+    /** @var TitleNode */
+    private $titleNode;
+
+    /** @var TemplateRenderer */
+    private $templateRenderer;
+
+    private static $idRenderedByFilename = [];
```

Comment:
This way tests are broken, only when full suite is run. I don't know how I can fix it. (except a static method to reset array, and call it before running phpunit test)

Reply:
This is always tricky. I'd prefer to not add a method to the class if it's only used in tests.

We could use reflection in the tests:

```php
$reflection = new \ReflectionClass(TitleNodeRenderer::class);
$property = $reflection->getProperty('idRenderedByFilename');
$property->setAccessible(true);
```